### PR TITLE
Added all_scores property to BeamSearchDecoderOutput to expose scores for all tokens

### PR DESCRIPTION
**Pull request by tgaddair**
# Description

The purpose of this PR is expose the scores for all token IDs, not only the ones that are selected as `predicted_ids`. In doing so, we enable the user of the `BeamSearchDecoder` to directly access the logits as requested in #1915.

This PR also fixes documentation from #1916 that described the `scores` property as having the shape of `all_scores`. Based on the existing unit tests, it looks as though the intent is for `scores` to have the shape `[batch_size, beam_width]`, not `[batch_size, beam_width, vocab_size]`, which is what is now exposed through `all_scores`.

## Type of change

- [ ] Bug fix
- [ ] New Tutorial
- [X] Updated or additional documentation
- [ ] Additional Testing
- [ ] New Activation and the changes conform to the [activation contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/activations/README.md#contribution-guidelines)
- [ ] New Callback and the changes conform to the [callback contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/callbacks/README.md#contribution-guidelines)
- [ ] New Image addition and the changes conform to the [image op contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/image/README.md#contribution-guidelines)
- [ ] New Layer and the changes conform to the [layer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/layers/README.md#contribution-guidelines)
- [ ] New Loss and the changes conform to the [loss contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/losses/README.md#contribution-guidelines)
- [ ] New Metric and the changes conform to the [metric contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/metrics/README.md#contribution-guidelines)
- [ ] New Optimizer and the changes conform to the [optimizer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/optimizers/README.md#contribution-guidelines)
- [ ] New RNN Cell and the changes conform to the [rnn contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/rnn/README.md#contribution-guidelines)
- [X] New Seq2seq addition and the changes conform to the [seq2seq contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/seq2seq/README.md#contribution-guidelines)
- [ ] New Text addition and the changes conform to the [text op contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/text/README.md#contribution-guidelines)

# Checklist:

- [X] I've properly [formatted my code according to the guidelines](https://github.com/tensorflow/addons/blob/master/CONTRIBUTING.md#coding-style)
    - [ ] By running Black + Flake8
    - [ ] By running pre-commit hooks
- [ ] This PR addresses an already submitted issue for TensorFlow Addons
- [X] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [ ] This PR contains modifications to C++ custom-ops

# How Has This Been Tested?

Unit tests have been added to verify that the new property has the correct shape.

### CODE REVIEW

**Comment by tgaddair** [comment]
@googlebot I signed it!

**Comment by tgaddair** [comment]
FYI, @guillaumekln @qlzh727 looks like failures are unrelated:

```
#17 [base_install 9/15] RUN python -m pip install --default-timeout=1000 ten...
#17 0.662 Collecting tensorflow==2.3.0
#17 1.925   Downloading https://files.pythonhosted.org/packages/fc/55/28d97d62ac476d589acdbf501f20dfb6265fbc016dcbd2e0e628d5b4b87a/tensorflow-2.3.0-cp35-cp35m-manylinux2010_x86_64.whl (320.4MB)
#17 4.233 THESE PACKAGES DO NOT MATCH THE HASHES FROM THE REQUIREMENTS FILE. If you have updated the package versions, please update the hashes. Otherwise, examine the package contents carefully; someone may have tampered with them.
#17 4.233     tensorflow==2.3.0 from https://files.pythonhosted.org/packages/fc/55/28d97d62ac476d589acdbf501f20dfb6265fbc016dcbd2e0e628d5b4b87a/tensorflow-2.3.0-cp35-cp35m-manylinux2010_x86_64.whl#sha256=6f74ef59dc59cf8f2002738c65dffa591e2c332e9b1b4ced33ff8d39b6fb477c:
#17 4.233         Expected sha256 6f74ef59dc59cf8f2002738c65dffa591e2c332e9b1b4ced33ff8d39b6fb477c
#17 4.233              Got        d061cfc74bce02ceecbf5f2f9d3459b562dc7afec6e9d0f2f5cc9e4bc05e3286
#17 4.233 
#17 4.358 You are using pip version 19.0, however version 20.2.1 is available.
#17 4.358 You should consider upgrading via the 'pip install --upgrade pip' command.
#17 ERROR: executor failed running [/bin/sh -c python -m pip install --default-timeout=1000 tensorflow==$TF_VERSION]: runc did not terminate sucessfully
```

**Comment by seanpmorgan** [comment]
> FYI, @guillaumekln @qlzh727 looks like failures are unrelated:
> 
> ```
> #17 [base_install 9/15] RUN python -m pip install --default-timeout=1000 ten...
> #17 0.662 Collecting tensorflow==2.3.0
> #17 1.925   Downloading https://files.pythonhosted.org/packages/fc/55/28d97d62ac476d589acdbf501f20dfb6265fbc016dcbd2e0e628d5b4b87a/tensorflow-2.3.0-cp35-cp35m-manylinux2010_x86_64.whl (320.4MB)
> #17 4.233 THESE PACKAGES DO NOT MATCH THE HASHES FROM THE REQUIREMENTS FILE. If you have updated the package versions, please update the hashes. Otherwise, examine the package contents carefully; someone may have tampered with them.
> #17 4.233     tensorflow==2.3.0 from https://files.pythonhosted.org/packages/fc/55/28d97d62ac476d589acdbf501f20dfb6265fbc016dcbd2e0e628d5b4b87a/tensorflow-2.3.0-cp35-cp35m-manylinux2010_x86_64.whl#sha256=6f74ef59dc59cf8f2002738c65dffa591e2c332e9b1b4ced33ff8d39b6fb477c:
> #17 4.233         Expected sha256 6f74ef59dc59cf8f2002738c65dffa591e2c332e9b1b4ced33ff8d39b6fb477c
> #17 4.233              Got        d061cfc74bce02ceecbf5f2f9d3459b562dc7afec6e9d0f2f5cc9e4bc05e3286
> #17 4.233 
> #17 4.358 You are using pip version 19.0, however version 20.2.1 is available.
> #17 4.358 You should consider upgrading via the 'pip install --upgrade pip' command.
> #17 ERROR: executor failed running [/bin/sh -c python -m pip install --default-timeout=1000 tensorflow==$TF_VERSION]: runc did not terminate sucessfully
> ```

Sorry about that it's a flaky failure related to GitHub Actions

**Comment by guillaumekln** [comment]
Thanks for the change. I'm a little bit concerned about memory usage, especially for users who do not need this information. Could it be optional in some way?

**Comment by tgaddair** [comment]
Hey @guillaumekln, thanks for taking a look.  I added a param to `BeamSearchDecoder` called `output_all_scores` that will toggle the `BeamSearchDecoderOutput.scores` property to contain either the top scores (default) or all scores.  Let me know if this addresses your concerns.  

I figured this would make more sense than keeping both properties, but having one be `None` unless an upstream param is set, which seems unintuitive.
